presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformCorrelatedSingleRowSubqueryToProject.java

```diff
@@ -95,14 +97,27 @@ public PlanNode visitLateralJoin(LateralJoinNode lateral, RewriteContext<PlanNod
             if (subqueryProjections.size() == 0) {
                 return rewrittenLateral.getInput();
             }
-            else if (subqueryProjections.size() == 1) {
-                Assignments assignments = Assignments.builder()
-                        .putIdentities(rewrittenLateral.getInput().getOutputSymbols())
-                        .putAll(subqueryProjections.get(0).getAssignments())
-                        .build();
-                return projectNode(rewrittenLateral.getInput(), assignments);
+
+            List<ProjectNode> reverseSubqueryProjection = new ArrayList<>(subqueryProjections);
+
+            Collections.reverse(reverseSubqueryProjection);
+
+            Assignments assignments = Assignments.builder().putIdentities(rewrittenLateral.getInput().getOutputSymbols())
```

Comment:
if you have a chained code like `obj.method1().method2()` please wrap lines to:
```
obj.method1()
    .method2()
```

Here:
```
Assignments.builder()
   .putIdentities(rewrittenLateral.getInput().getOutputSymbols())
```